tests/test_threaded_backend.py

```diff
@@ -131,6 +137,19 @@ def test_worker_error(self):
         self.assertTrue(self.backend.thread is None)
 
 
+class TestThreaded(object):
+    def test_calling_thread_waits_for_worker(self):
+        """
+        Start a process we are not the parent of and see if it waits for at
+        least the queue timeout (1) before exiting. If we were using a daemon
+        thread the process would exit immediately.
+        """
+        start = time.time()
```

Comment:
I think this test is a good thing, since it actually tests a main thread, but the assertion is implicit (assuming a max execution time of 1 sec).

Not a blocker for me but, ideally, we should also have a test with assertions for the threads state (alive and stopped) in the proper sequence. And for this test, it doesn't have to be with a main thread.